Add deletetask command

w-yeehong opened the pull request:
1. Add `deletetask` command. It takes a room number and a task number. E.g. `deletetask r/2 t/1` would delete the first task from room number 2.

2. Add `toString()` method to `TaskList`. This method will be used to display task information in the UI.

3.  Update UG with instructions on how to use `deletetask`.

w-yeehong committed:
TaskList: Add toString method

w-yeehong committed:
AddTaskCommandParser: Add Javadoc

w-yeehong committed:
Support task deletion from Room and RoomList

w-yeehong committed:
Add integration tests for Room and RoomList task deletion

w-yeehong committed:
Add deletetask command

w-yeehong committed:
Add unit tests for addtask and deletetask parsing

w-yeehong committed:
Remove unnecessary unit tests

w-yeehong committed:
Add unit tests for deletetask command

w-yeehong committed:
UserGuide.md: Add instructions on using deletetask

raymondge commented on src/main/java/seedu/address/logic/commands/task/DeleteTaskCommand.java:
```diff
@@ -0,0 +1,81 @@
+            throw new CommandException(Messages.MESSAGE_INVALID_ROOM_INDEX);
+        }
+        Room targetRoom = rooms.get(roomIndex.getZeroBased());
+        assert targetRoom != null : "Target room should never be null.";
```
good assertion! If the targetroom is null, it will be prehandled by the commandexception

raymondge commented on src/main/java/seedu/address/logic/commands/task/DeleteTaskCommand.java:
```diff
@@ -0,0 +1,81 @@
+        Task taskToDelete = tasks.get(taskIndex.getZeroBased());
+        assert taskToDelete != null : "The task to delete should never be null.";
+
+        model.deleteTaskFromRoom(taskToDelete, targetRoom);
```
I think instead of letting the model to handle deleteTask, the deletetaskcommand class should handle it instead for abstraction

w-yeehong replied:
Hmm could you clarify on this? I think most of the other `Command` classes handle add, delete, and update through `Model`. The `Model` is the abstraction.

raymondge replied:
oh then that's ok I guess

raymondge commented on src/test/java/seedu/address/logic/commands/task/DeleteTaskCommandTest.java:
```diff
@@ -0,0 +1,54 @@
+    public void constructor_validRoomIndexNullTaskIndex_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new DeleteTaskCommand(VALID_ROOM_INDEX_ONE, null));
+    }
+
```
Maybe add a success case for test?

chiamyunqing replied:
Seems like success cases are covered in equals() method test.

w-yeehong replied:
It is missing the success case actually. Will need to set up a stub to perform the test. Will KIV 1st.

raymondge reviewed (approved):
A lot of hard work! LGTM!

chiamyunqing commented on src/main/java/seedu/address/logic/parser/task/TaskCliSyntax.java:
```diff
@@ -9,7 +9,8 @@
     public static final Prefix PREFIX_DESCRIPTION = new Prefix("d/");
-    public static final Prefix PREFIX_ROOM_NUMBER = new Prefix("r/");
     public static final Prefix PREFIX_DUE_DATE = new Prefix("dd/");
+    public static final Prefix PREFIX_ROOM_NUMBER = new Prefix("r/");
```
Is it better to use the prefix_room_number from RoomCliSyntax instead since room number is an attribute of room?

w-yeehong replied:
Good point. Will use that instead 👍

chiamyunqing reviewed (approved):
Clean and neat code with comprehensive test coverage. Well done!

w-yeehong committed:
Replace getRoomDisplayRoom() method with getRoomList()

w-yeehong committed:
RoomDetailsPanel: Print task details with TaskList#toString()

The two methods TaskList#toString() and TaskList#iterateTaskDetails()
display overlapping behaviors. Both are used to print task details.

Let's consolidate the two methods into TaskList#toString() instead
to avoid duplicate code.
